src/components/InteractiveMap/InteractiveMap.vue

```diff
@@ -105,105 +52,125 @@
-/// <reference path="@/types/escher.d.ts" />
-import * as escher from "@dd-decaf/escher";
 import * as settings from "@/settings";
+import Escher from "@/components/InteractiveMap/Escher.vue";
+import Card from "@/components/InteractiveMap/Card.vue";
 
 export default Vue.extend({
   name: "InteractiveMap",
+  components: {
+    Escher,
+    Card
+  },
   data: () => ({
     escherBuilder: null,
-    isInitializingEscher: true,
     currentMapId: null,
-    isLoadingMap: false,
-    cards: [
-      {
-        // TODO: Replace test card
-        name: "Design",
-        method: "pFBA",
-        model: { name: "iJO1366" },
-        organism: { name: "E. coli" },
-        growthRate: 0.874
-      }
-    ]
+    fluxDistribution: null,
+    mapData: null,
+    cards: []
   }),
   methods: {
+    escherLoaded() {
+      this.$store.getters["models/onData"](this.addDefaultCard);
+    },
     changeMap() {
-      this.isLoadingMap = true;
       // TODO: Get map from maps state lazy loader
       axios
         .get(`${settings.apis.maps}/maps/${this.currentMapId}`)
         .then(response => {
-          // Note that this will freeze the entire application, including
-          // progress spinners.
-          this.escherBuilder.load_map(response.data.map);
-          this.isLoadingMap = false;
+          this.mapData = response.data.map;
+        })
+        .catch(error => {
+          // TODO: show snackbar
         });
     },
-    reactionState(id: string, type?: string) {
-      // TODO
-    },
-    handleKnockout(reactionId: string) {
-      // TODO
-    },
-    handleKnockoutGenes(reactionId: string) {
-      // TODO
+    addDefaultCard() {
+      const model = this.$store.state.models.models.find(
+        model => model.name === "e_coli_core"
+      );
+      if (model) {
+        const organism = this.$store.getters["organisms/getOrganismById"](
+          model.organism_id
+        );
+        this.addCard("Design", organism, model, "pfba");
+      } else {
+        this.addCard("Design", null, null, "pfba");
+      }
     },
-    handleSetAsObjective(reactionId: string) {
-      // TODO
+    addCard(name, organism, model, method) {
+      const card = {
+        name: name,
+        organism: organism,
+        model: model,
+        method: method,
+        isSimulating: false,
+        growthRate: null,
+        isSelected: false
+      };
+      this.cards.push(card);
+      this.selectCard(card);
+      this.simulate(card);
     },
-    handleChangeBounds(reactionId: string, lower: string, upper: string) {
-      // TODO
+    removeCard(card) {
+      this.cards.splice(this.cards.indexOf(card), 1);
     },
-    handleResetBounds(reactionId: string) {
-      // TODO
+    selectCard(card) {
+      this.cards.forEach(card => {
+        card.isSelected = false;
+      });
+      card.isSelected = true;
     },
-    handleObjectiveDirection(reactionId: string) {
-      // TODO
+    simulate(card) {
+      if (card.model === null) {
```

Comment:
If necessary you can change the comparison to below in order to capture both `null` and `undefined`.

```suggestion
      if (card.model == null) {
```